p2p/net/swarm/swarm_notif_test.go

```diff
@@ -0,0 +1,186 @@
+package swarm
+
+import (
+	"testing"
+	"time"
+
+	inet "github.com/jbenet/go-ipfs/p2p/net"
+
+	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/go.net/context"
+)
+
+func TestNotifications(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	swarms := makeSwarms(ctx, t, 5)
+	defer func() {
+		for _, s := range swarms {
+			s.Close()
+		}
+	}()
+
+	timeout := 5 * time.Second
+
+	// signup notifs
+	notifiees := make([]*netNotifiee, len(swarms))
+	for i, swarm := range swarms {
+		n := newNetNotifiee()
+		swarm.Notify(n)
+		notifiees[i] = n
+	}
+
+	connectSwarms(t, ctx, swarms)
+
+	<-time.After(time.Millisecond)
```

Comment:
doing something time based in a test with a resolution of one Millisecond feels janky to me